packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -633,3 +639,596 @@ describe('when dispatching a loading notification', () => {
+    fireEvent.click(button);
+    await wait(() => {
+      expect(window.localStorage.setItem).toHaveBeenCalledWith(
+        STORAGE_KEYS.IS_FORCED_MENU_OPEN,
+        false
+      );
+    });
+    fireEvent.click(button);
+    await wait(() => {
+      expect(window.localStorage.setItem).toHaveBeenCalledWith(
+        STORAGE_KEYS.IS_FORCED_MENU_OPEN,
+        true
+      );
+    });
+    // Check that the support link is rendered
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+    expect(navbarRendered.queryByText('Support')).toBeInTheDocument();
+  });
+});
+describe('when rendering navbar menu links from local config', () => {
+  let operations;
+  beforeEach(() => {
+    operations = mocksForMc.createMockOperations();
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: operations },
+    });
+    selectProjectKeyFromUrl.mockReturnValue(
+      operations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should render links with all the correct state attributes', async () => {
+    const navbarSubmenuMock = ApplicationNavbarSubmenuMock.build();
+    const navbarMock = ApplicationNavbarMenuMock.build({
+      submenu: [navbarSubmenuMock],
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+
+    const applicationLocale = operations.FetchLoggedInUser.me.language;
+    const mainMenuLabel = navbarMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    const mainSubmenuLabel = navbarSubmenuMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    // Check the relationships between the menu items of a group
+    const menuTitle = await navbarRendered.findByText(mainMenuLabel.value);
+    const groupId = menuTitle.getAttribute('aria-owns');
+    expect(rendered.container.querySelector(`#${groupId}`)).toHaveAttribute(
+      'aria-expanded',
+      'false'
+    );
+    fireEvent.click(menuTitle);
+    expect(rendered.container.querySelector(`#${groupId}`)).toHaveAttribute(
+      'aria-expanded',
+      'true'
+    );
+    const menuGroup = within(rendered.getByTestId(groupId));
+    expect(menuGroup.queryByText(mainSubmenuLabel.value)).toBeInTheDocument();
+    expect(menuGroup.queryByText(mainSubmenuLabel.value)).not.toHaveAttribute(
+      'aria-current'
+    );
+    // Go to the link to check if the link becomes active
+    fireEvent.click(menuGroup.queryByText(mainSubmenuLabel.value));
+    await wait(() => {
```

Comment:
what is the reason for the `wait` here and not having that for L717?

Reply:
Hmm no reason, I can maybe await above as well. Thanks for pointing it out

Reply:
hmm.. if there is no reason to do so here and above, then don't :)
my intention was to understand the workflow of the menu that requires that we use `wait`